### DISCUSSION TITLE
showPins() not showing refreshed pins after pin create, delete or update

### DISCUSSION THREAD

**Discussion by GonzalesMatthew** [nss-evening-cohort-14/evening-client-side#175] [Help Me!] [answered]
Your Name: Matthew Gonzales

Have you searched the other help tickets?
Yes.

My showPins() function works when clicking on a board to view, but when creating, deleting, or updating a pin (which do work on the back end), it doesn't refresh my view and I have to manually click through to the board pins again.

List any error messages that appear
No error message. But I think my pinsArray is empty after console.warn-ing it, but not sure why.

What have you googled?
I don't think this is something I can google. I believe it's a routing issue, or I'm technically doing something wrong. I'm not getting any console errors.

What have you tried? Tell us everything you have tried.
I navigated through all my functions and domEvents and everything looks correct.
I double-checked my id's, but perhaps I'm missing something here. Still things look correct.
I console.warned the response from the axios.get command and I'm getting an empty array, but not sure why.

```
const updatePin = (firebaseKey, pinObject) => new Promise((resolve, reject) => {
  axios.patch(`${dbUrl}/pins/${firebaseKey}.json`, pinObject)
    .then(() => getBoardsPins(firebase.auth().currentUser.uid))
    // .then((pinsArray) => resolve(pinsArray))
    .then((pinsArray) => console.warn(pinsArray))
    .catch((error) => reject(error));
});
```
Link to my repo
https://github.com/GonzalesMatthew/ASSIGNMENT-Firebase-Pinterest

Thank you!

**Comment by TrinityChristiana** [comment, marked as answer]
Looks like `getBoardsPins` is expecting a `boardId` but you're passing it `firebase.auth().currentUser.uid`

`getBoardsPins` definition: https://github.com/GonzalesMatthew/ASSIGNMENT-Firebase-Pinterest/blob/main/src/javascripts/helpers/data/pinData.js#L10-L14

Where you're calling it in `updatePin`: https://github.com/GonzalesMatthew/ASSIGNMENT-Firebase-Pinterest/blob/main/src/javascripts/helpers/data/pinData.js#L45

**Reply by GonzalesMatthew** [reply]
You're right! I'm trying to refactor my create/delete/update to look at the pinObject.board_id instead now. Create is working! Working on update and delete. Thanks, Trin!
